components/samsung_ac/protocol_nasa.cpp

```diff
@@ -556,55 +569,88 @@ namespace esphome
             case MessageNumber::VAR_in_temp_room_f: //  unit = 'Celsius' from XML
             {
                 double temp = (double)message.value / (double)10;
-                ESP_LOGW(TAG, "s:%s d:%s VAR_in_temp_room_f %f", source.c_str(), dest.c_str(), temp);
+                if (debug_log_messages)

```

Comment:
Adding the debug_log_message if to each case makes this already too large method even more larger. Can you please add and LOG_MESSAGE marco for that case?